runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/client/grpc/GrpcCommitWorkStream.java

```diff
@@ -314,26 +418,43 @@ private Batcher() {
     @Override
     public boolean commitWorkItem(
         String computation, WorkItemCommitRequest commitRequest, Consumer<CommitStatus> onDone) {
+      if (isShutdown()) {
+        onDone.accept(CommitStatus.ABORTED);
+        return false;
+      }
+
       if (!canAccept(commitRequest.getSerializedSize() + computation.length())) {
         return false;
       }
-      PendingRequest request = new PendingRequest(computation, commitRequest, onDone);
+
+      PendingRequest request = PendingRequest.create(computation, commitRequest, onDone);
       add(idGenerator.incrementAndGet(), request);
       return true;
     }
 
     /** Flushes any pending work items to the wire. */
     @Override
     public void flush() {
-      flushInternal(queue);
-      queuedBytes = 0;
-      queue.clear();
+      try {
+        if (!isShutdown()) {
+          flushInternal(queue);
+        } else {
+          queue.forEach((ignored, request) -> request.abort());
+        }
+      } finally {
+        queuedBytes = 0;
+        queue.clear();
+      }
     }
 
     void add(long id, PendingRequest request) {
-      assert (canAccept(request.getBytes()));
-      queuedBytes += request.getBytes();
-      queue.put(id, request);
+      if (isShutdown()) {
```

Comment:
seems like this should just be in commitWorkItem or here not both

Reply:
removed